Model/LibBookmarks/LibBookmarksBridge.mm

```diff
@@ -0,0 +1,56 @@
+//
+//  LibBookmarksBridge.mm
+//  Kiwix
+
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdocumentation"
+#include "kiwix/library.h"
+#include "kiwix/bookmark.h"
+#pragma clang diagnostic pop
+
+#import "LibBookmarksBridge.h"
+
+# pragma mark - Private properties
+@interface LibBookmarksBridge ()
+
+@property kiwix::LibraryPtr library;
+
+@end
+
+
+# pragma mark - Implementation
+
+@implementation LibBookmarksBridge
+
+- (instancetype _Nonnull)init {
+    self = [super init];
+    if (self) {
+        self.library = kiwix::Library::create();
+    }
+    return self;
+}
+
+- (BOOL) isBookmarked: (nonnull NSURL *) url inZIM: (nonnull NSUUID *) zimFileID {
+    std::string url_c = [url fileSystemRepresentation];
+    std::string fileID_c = [[[zimFileID UUIDString] lowercaseString] cStringUsingEncoding:NSUTF8StringEncoding];
+    const std::vector<kiwix::Bookmark> bookmarks = self.library->getBookmarks(true);
```

Comment:
if this call can be used without the argument and it works equally good:
`self.library->getBookmarks();`
The tricky part is, which is not much documented, is that the `kiwix::Book` needs to be added to the library,
as on line 47.
If the `Book` is not added to the `library`, the `self.library->getBookmarks();` won't return it, as it is an "invalid" bookmark..
The other solution is not to add the Book to the library and call `getBookmarks(false)`.
Well... without looking into the C++ code it is not obvious...